app/services/list_tasks.rb

```diff
@@ -0,0 +1,24 @@
+class ListTasks < ListCollection
+
+  attr_defaultable :task_respository, -> { Task }
+  attr_defaultable :result_serializer, -> { V1::TaskSerializer }
+
+
+  attr_reader :project_id
+
+
+  def initialize project_id: nil, page: 1, page_size: 25
+    super()
+    @project_id = project_id
+    raise ArgumentError, 'Project not found' if @project_id.empty?
```

Comment:
It's good to check you are getting a project id, but it seems just checking for the presence of the id is enough to know if you have a project so your error message `Project not found` seems out of step with what you are checking.